rust/ql/test/utils/InlineFlowTest.qll

```diff
@@ -0,0 +1,37 @@
+/**
+ * Inline flow tests for Rust.
+ * See `shared/util/codeql/dataflow/test/InlineFlowTest.qll`
+ */
+
+import rust
+private import codeql.dataflow.test.InlineFlowTest
+private import codeql.rust.dataflow.DataFlow
+private import codeql.rust.dataflow.internal.DataFlowImpl
+private import codeql.rust.dataflow.internal.TaintTrackingImpl
+private import internal.InlineExpectationsTestImpl as InlineExpectationsTestImpl
+
+private module FlowTestImpl implements InputSig<Location, RustDataFlow> {
+  predicate defaultSource(DataFlow::Node source) {
+    source.asExpr().(CallExpr).getExpr().toString() = "source"
+  }
+
+  predicate defaultSink(DataFlow::Node sink) {
+    any(CallExpr call | call = sink.asExpr() and call.getExpr().toString() = "sink")
+        .getArgList()
+        .getAnArg() = sink.asExpr()
+  }
+
+  private string getSourceArgString(DataFlow::Node src) {
+    defaultSource(src) and
+    result = src.asExpr().(CallExpr).getArgList().getArg(0).toString()
+  }
+
+  string getArgString(DataFlow::Node src, DataFlow::Node sink) {
```

Comment:
Missing `bindingset[src, sink]`.

Reply:
What does the `bindingset` do here? From the documentation I thought that it was only to be used for predicates over infinite types. For several of the other language there is no `bindingset` annotation on `getArgString`. [Go for instance](https://github.com/paldepind/codeql/blob/d9d3027667e19e98816c731bcb0dfefec0e0564c/go/ql/test/TestUtilities/InlineFlowTest.qll?plain=1#L24). Should it ideally also be added there? Or is it not needed there for some reason?

Reply:
It is needed to avoid creating a Cartesian product. It should also be added for Go.